src/helpers/useFocusItemHelper.ts

```diff
@@ -0,0 +1,171 @@
+   */
+  innerSpacing?: number;
+};
+
+// TODO: this is what I want, is there a better way to do it?
+type Offsets = {
+  CENTER: number[];
+  LEFT: number[];
+  RIGHT: number[];
+};
+
+export type ResultProps = {
+  /**
+   * This should be called by each ot the items' onLayout
+   */
+  onItemLayout: (event: LayoutChangeEvent, index: number) => void;
+  /**
+   * The items' width
+   */
+  itemsWidths: number[];
+  /**
+   * Use in order to focus the item with the specified index
+   */
+  focusIndex: (index: number, animated?: boolean) => void;
+};
+
+const useFocusItemsHelper = (props: Props): ResultProps => {
+  const {
+    scrollViewRef,
+    itemsCount,
+    selectedIndex,
+    offsetType = OffsetType.CENTER,
+    addOffsetMargin = true,
+    sideSpacing = 0,
+    innerSpacing = 0
+  } = props;
+  const itemsWidths = useRef<(number | null)[]>(_.times(itemsCount, () => null));
+  const currentIndex = useRef<number>(selectedIndex || 0);
+  const [offsets, setOffsets] = useState<Offsets>({CENTER: [], LEFT: [], RIGHT: []});
+
+  // TODO: reset?
+  //   useEffect(() => {
+  //     itemsWidths.current = _.times(itemsCount, () => null);
+  //   }, [itemsCount]);
+
+  // const contentWidth = _.sum(itemsWidths);
+  // TODO: const scrollEnabled = contentWidth.current > containerWidth;
+
+  const setSnapBreakpoints = useCallback((itemsWidths: number[]) => {
+    if (_.isEmpty(itemsWidths)) {
+      return;
+    }
+
+    const screenCenter = Constants.screenWidth / 2; // TODO: change to something more dynamic?
+    let index = 0;
+    const centeredOffsets = [];
+    let currentCenterOffset = sideSpacing;
+    const leftOffsets = [];
+    leftOffsets.push(0);
+    const rightOffsets = [];
+    rightOffsets.push(-Constants.screenWidth + itemsWidths[0]);
+    while (index < itemsCount) {
+      centeredOffsets[index] = currentCenterOffset - screenCenter + itemsWidths[index] / 2;
+      ++index;
+      currentCenterOffset += itemsWidths[index - 1] + innerSpacing;
+      leftOffsets[index] = leftOffsets[index - 1] + itemsWidths[index - 1];
+      rightOffsets[index] = rightOffsets[index - 1] + itemsWidths[index];
+    }
+
+    if (addOffsetMargin) {
+      index = 1;
+      while (index < itemsCount - 1) {
+        leftOffsets[index] -= itemsWidths[index - 1];
+        rightOffsets[index] += itemsWidths[index + 1];
+        ++index;
+      }
+    }
+
+    setOffsets({CENTER: centeredOffsets, LEFT: leftOffsets, RIGHT: rightOffsets}); // default for DYNAMIC is CENTER
+  },
+  [itemsCount, sideSpacing, innerSpacing, addOffsetMargin]);
+
+  const onItemLayout = useCallback((event: LayoutChangeEvent, index: number) => {
+    const {width} = event.nativeEvent.layout;
+    itemsWidths.current[index] = width;
+    if (!_.includes(itemsWidths.current, null)) {
+      setSnapBreakpoints(itemsWidths.current as number[]);
+    }
+  },
+  [setSnapBreakpoints]);
+
+  const scroll = (scrollTo: number, animated: boolean) => {
```

Comment:
Is this in order to support ScrollView, FlatList and SectionList? 
I think I have a similar code somewhere.. (: 

Maybe we should add a specific hook for that (:
Anyway, not for this scope - just FYI

Reply:
No time like the present.
I'm just not sure it's a hook, maybe a helper function, since it doesn't need `useState`, `useRef` etc, WDYT?

Reply:
Technically you can have this hook to create the scrollView ref and provide (and also use it internally for the scrollTo)
something like that.. 
```
const [scrollRef, scrollTo] = useScrollTo()
```
and maybe if someone already has the ref we can support passing it 
```
const myOwnRef = useRef();
const [scrollRef, scrollTo] = useScrollTo(myOwnRef)
```

WDYT?

Reply:
Done